client.go

```diff
@@ -285,6 +285,9 @@ type Client struct {
 	// PagerDuty API. You can use either *http.Client here, or your own
 	// implementation.
 	HTTPClient HTTPClient
+
+	// UserAgent allows to overwrite Client User Agent value
+	UserAgent string
```

Comment:
1) Could we please use the `ClientOptions` / `WithUserAgent` pattern for setting this value? (See `WithAPIEndpoint(endpoint string) ClientOptions` for an example of this.

2) Having the User-Agent based usage data is pretty valuable. What do you think about limiting the ability for clients to customize this? This might look something like:
```go
func WithTerraformProvider(version string) ClientOptions {
	return func(c *Client) {
		c.userAgent = fmt.Sprintf("Terraform/%s", version)
	}
}
```

Reply:
fair enough 👍🏽